extensions/register-and-verify-3.3.md

```diff
@@ -0,0 +1,196 @@
+network to provide a common interface regardless of what authentication layer they have chosen
+for their network to operate, i.e. traditional services, a different central authority, or a
+decentralized model similar to [IRCX][ircx].
+
+   [ircx]: http://en.wikipedia.org/wiki/IRCX
+   
+We believe that the benefit of adding a common interface for account creation and validation,
+is helpful to the end user in discovering the capabilities offered by the authentication and
+authority layer of the network.  Further, we believe it is helpful to client authors, so that
+they may provide guided account creation and validation for their users, regardless of network
+configuration.
+
+## The `REGISTER` command
+
+The `REGISTER` command signals the intent of a client to register an account in the network's
+authentication layer.  It is similar to current methods of signalling that intent.
+
+A `REGISTER` command consists of the following format:
+
+`REGISTER <accountname> <callback> <cred_type> :<credential>`
+
+A passphrase `credential` MAY have spaces in it.
+
+The `callback` field designates an opaque value which indicates where a verification code, if any,
+will be sent.  The callback field is implementation-specific, with namespaces being listed using the
+`REGCALLBACKS` ISUPPORT token.  An invalid callback parameter should result in the `ERR_REG_INVALID_CALLBACK`
+error.  In the event that a callback is not provided, the client SHOULD send `*` to indicate that
+they are not providing a callback resource.
+
+The `cred_type` field designates a value which indicates the type of supplied credential.  The accepted
+values of the `cred_type` field is implementation-specific, with credential types being listed using the
+`REGCREDTYPES` ISUPPORT token.  An invalid credential type parameter should result in the `ERR_REG_INVALID_CRED_TYPE`
+error.  The client MUST supply a `cred_type` value.
+
+The IRC server MAY forward the `REGISTER` command to a central authority, or process it locally, however
+the authentication layer SHOULD NOT send any legacy messages in reply to a `REGISTER` command.
+
+Upon success, the IRC server MUST send the `RPL_REGISTRATION_SUCCESS` numeric, which looks like:
+
+    :<server> 920 <user_nickname> <accountname> :Account registered
+
+If the server requires a verification token, it must also reply with the `RPL_REG_VERIFICATION_REQUIRED`
+numeric, which looks like:
+
+    :<server> 927 <user_nickname> <accountname> <callback> :A verification token was sent
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these numerics:
+
+`ERR_ACCOUNT_ALREADY_EXISTS`:
+    `:<server> 921 <user_nickname> <accountname> :Account already exists`
+`ERR_REG_UNSPECIFIED_ERROR`:
+    `:<server> 922 <user_nickname> <accountname> :<descriptive_text>`
+`ERR_REG_INVALID_CALLBACK`:
+    `:<server> 926 <user_nickname> <accountname> <callback> :Callback token is invalid`
```

Comment:
@SaberUK can you provide a list of numerics in the area that are already used so I can assign one that does not have a collision?